workflows/mag.nf

```diff
@@ -487,25 +488,56 @@ workflow MAG {
 
     /*
     ================================================================================
-                                    Binning
+                                Binning preparation
     ================================================================================
     */
 
+
     ch_bowtie2_assembly_multiqc = Channel.empty()
     ch_busco_summary            = Channel.empty()
     ch_busco_multiqc            = Channel.empty()
 
+
+
+    BINNING_PREPARATION (
+        ch_assemblies,
+        ch_short_reads
+    )
+
+    /*
+    ================================================================================
+                                    Ancient DNA
+    ================================================================================
+    */
+
+    if (params.ancient_dna){
+        ANCIENT_DNA_ASSEMLY_VALIDATION(BINNING_PREPARATION.out.grouped_mappings)
+        ch_software_versions = ch_software_versions.mix(ANCIENT_DNA_ASSEMLY_VALIDATION.out.versions.ifEmpty(null))
+    }
+
+    /*
+    ================================================================================
+                                    Binning
+    ================================================================================
+    */
+
     if (!params.skip_binning){
 
-        BINNING_PREPARATION (
-            ch_assemblies,
-            ch_short_reads
-        )
+        if (params.ancient_dna) {
+            METABAT2_BINNING (
+                BINNING_PREPARATION.out.grouped_mappings
+                    .join(ANCIENT_DNA_ASSEMLY_VALIDATION.out.contigs_recalled)
+                    .map{ it -> [ it[0], it[4], it[2], it[3] ] }, // [meta, contigs_recalled, bam, bais]
+                ch_short_reads
+            )
```

Comment:
For readability reasons I don't particularly like nesting channel manipulation within module declarations, but rather prepare the channel outside the module declaration, and supply just that (but that's just me)

Reply:
I'll leave as is for now, as this is what allows to pipe into the next step more easily.